includes/core/server/request/class-paginated.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Paginated.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+
+/**
+ * Request class for paginated requests.
+ */
+class Paginated extends Request {
+
+
+	const DEFAULTS = [
+		'page'      => 0,
+		'pagesize'  => 25,
+		'sort'      => 'created',
+		'direction' => 'desc',
+		'limit'     => 100,
+	];
+
+	/**
+	 * Request uri
+	 *
+	 * @var mixed (int|string)
+	 */
+	private $uri = null;
+
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function get_api(): string {
+		return $this->uri; // Will be changed on every request by overriding this method or use set_uri method.
+	}
+
+	/**
+	 * Returns the request's HTTP method.
+	 */
+	public function get_method(): string {
+		return 'GET';
+	}
+
+	/**
+	 * Return exact same response from API server. In some cases, this is all we need.
+	 *
+	 * @param mixed $response Response from server.
+	 *
+	 * @return mixed|\WCPay\Core\Server\Response
+	 */
+	public function format_response( $response ) {
+		return $response;
+	}
+
+
+	/**
+	 * Used to provide custom check for setting filter keys. Some requests might have their own logic of which filters keys are mutable.
+	 *
+	 * @param string $key Key to check.
+	 *
+	 * @return bool
+	 */
+	public function is_filter_key_mutable( string $key ) {
+		// In most cases, the default parameters are keys that cannot be changed via filters.
+		return ! array_key_exists( $key, self::DEFAULTS );
+	}
+
+	/**
+	 * Set filters.
+	 *
+	 * @param array $filters Filterd to set.
+	 *
+	 * @return void
+	 */
+	public function set_filters( array $filters ) {
```

Comment:
Why not make the setters explicit, and add setters for individual filters within request classes?

Reply:
These filters have been added.